### PR TITLE
[SPARK-42273][CONNECT][TESTS] Skip Spark Connect tests if dependencies are not installed

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to skip doctests if the dependencies for Spark Connect are not installed.
The skipping logic has to be added before the modules are loaded because it loads the module first when Python runs doctest. Otherwise, it fails as below:

```
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.9/3.9.10/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/Cellar/python@3.9/3.9.10/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/.../spark/python/pyspark/sql/tests/connect/test_connect_basic.py", line 29, in <module>
    from pyspark.sql.connect.client import Retrying
  File "/.../spark/python/pyspark/sql/connect/__init__.py", line 21, in <module>
    from pyspark.sql.connect.dataframe import DataFrame  # noqa: F401
  File "/.../spark/python/pyspark/sql/connect/dataframe.py", line 50, in <module>
    import pyspark.sql.connect.plan as plan
  File "/.../spark/python/pyspark/sql/connect/plan.py", line 26, in <module>
    import pyspark.sql.connect.proto as proto
  File "/.../spark/python/pyspark/sql/connect/proto/__init__.py", line 18, in <module>
    from pyspark.sql.connect.proto.base_pb2_grpc import *
  File "/.../spark/python/pyspark/sql/connect/proto/base_pb2_grpc.py", line 19, in <module>
    import grpc
ModuleNotFoundError: No module named 'grpc'
```

### Why are the changes needed?

To make the tests pass without these optional dependencies.

### Does this PR introduce _any_ user-facing change?

No, dev-only.

### How was this patch tested?

```
./python/run-tests --module pyspark-connect -p 1
```

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @zhengruifeng

**Review by zhengruifeng — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @HyukjinKwon !

File: python/pyspark/sql/connect/__init__.py [hunk 1]
```diff
@@ -17,14 +17,26 @@
...
     require_minimum_pandas_version,
     require_minimum_pyarrow_version,
     require_minimum_grpc_version,
 )
 
-require_minimum_pandas_version()
-require_minimum_pyarrow_version()
-require_minimum_grpc_version()
+
+def check_dependencies(mod_name, file_name):
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/__init__.py:29]:
```suggestion
def check_dependencies(mod_name: str, file_name: str) -> None:
```

Comment (HyukjinKwon) [inline python/pyspark/sql/connect/__init__.py:29]:
```suggestion
def check_dependencies(mod_name: str, file_name: str) -> None:
```

**Comment by HyukjinKwon** [comment]
The last change was just to fix the type hint that dosen;t affect the test results.

Merged to master and branch-3.4.

**Pull request closed without merging**